File: kurs-backend/src/main/java/com/example/vkr/entity/Metric.java
package com.example.vkr.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "metrics")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Metric {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "device_id", nullable = false)
    private Device device;

    private Double cpuLoad;
    private Double ramUsage;
    private Double diskUsage;
    private Long uptime;
    private LocalDateTime timestamp;
}

File: kurs-frontend/src/app/models/metric.ts
export interface Metric {
  id: number;
  deviceId: number;
  cpuLoad: number;
  ramUsage: number;
  diskUsage: number;
  uploadMbps?: number;
  downloadMbps?: number;
  pingMs?: number;
  packetLossPercent?: number;
  uptime?: number;
  timestamp: string;
}
